wrangler-service/src/main/java/co/cask/wrangler/service/bigquery/BigQueryService.java

```diff
@@ -0,0 +1,347 @@
+    }
+
+    if (properties.get(SERVICE_ACCOUNT_KEYFILE) == null) {
+      throw new Exception("Configuration does not include path to service account file.");
+    }
+
+    String path = (String) properties.get(SERVICE_ACCOUNT_KEYFILE);
+    String projectId = (String) properties.get(PROJECT_ID);
+    ServiceAccountCredentials credentials = GCPServiceAccount.loadLocalFile(path);
+
+    return BigQueryOptions.newBuilder()
+      .setProjectId(projectId)
+      .setCredentials(credentials)
+      .build()
+      .getService();
+  }
+
+  /**
+   * Tests GCS Connection.
+   *
+   * @param request HTTP Request handler.
+   * @param responder HTTP Response handler.
+   */
+  @POST
+  @Path("/connections/bigquery/test")
+  public void testBiqQueryConnection(HttpServiceRequest request, HttpServiceResponder responder) {
+    try {
+      // Extract the body of the request and transform it to the Connection object.
+      RequestExtractor extractor = new RequestExtractor(request);
+      Connection connection = extractor.getContent(Charsets.UTF_8.name(), Connection.class);
+      ConnectionType connectionType = ConnectionType.fromString(connection.getType().getType());
+      if (connectionType == ConnectionType.UNDEFINED || connectionType != ConnectionType.BIGQUERY) {
+        error(responder,
+              String.format("Invalid connection type %s set, expected 'BIGQUERY' connection type.",
+                            connectionType.getType()));
+        return;
+      }
+
+      BigQuery bigQuery = getBigQuery(connection);
+      bigQuery.listDatasets(BigQuery.DatasetListOption.pageSize(1));
+      ServiceUtils.success(responder, "Success");
+    } catch (Exception e) {
+      ServiceUtils.error(responder, e.getMessage());
+    }
+  }
+
+  /**
+   * List all Datasets.
+   *
+   * @param request HTTP requets handler.
+   * @param responder HTTP response handler.
+   */
+  @GET
+  @Path("connections/{connection-id}/bigquery")
+  public void listDatasets(HttpServiceRequest request, HttpServiceResponder responder,
+                           @PathParam("connection-id") String connectionId) throws Exception {
+    Connection connection = store.get(connectionId);
+
+    if (!validateConnection(connectionId, connection, responder)) {
+      return;
+    }
+    BigQuery bigQuery = getBigQuery(connection);
+    Page<Dataset> datasets = bigQuery.listDatasets(BigQuery.DatasetListOption.all());
+    List<JsonObject> allDatasets = new ArrayList<>();
+    for (Dataset dataset : datasets.iterateAll()) {
+      JsonObject object =  new JsonObject();
+      object.addProperty("name", dataset.getDatasetId().getDataset());
+      object.addProperty("created", dataset.getCreationTime());
+      object.addProperty("description", dataset.getDescription());
+      object.addProperty("last-modified", dataset.getLastModified());
+      object.addProperty("location", dataset.getLocation());
+      allDatasets.add(object);
+    }
+    responder.sendJson(allDatasets);
```

Comment:
Response is not inline with other responses being sent. Please make them the same.